torch/_inductor/codegen/triton.py

```diff
@@ -832,42 +845,35 @@ def constant(cls, value, dtype):
 
     @staticmethod
     def abs(x):
-        return f"tl_math.abs({x})"
+        return _maybe_upcast_float32("tl_math.abs", [x])
 
     @staticmethod
     def libdevice_abs(x):
-        return f"libdevice.abs({x})"
+        return _maybe_upcast_float32("libdevice.abs", [x])
 
     @staticmethod
     def exp(x):
-        return f"tl_math.exp({x})"
+        return _maybe_upcast_float32("tl_math.exp", [x])
```

Comment:
nit - would this be cleaner as a decorator ? 
```
@maybe_upcast_float32
@staticmethod
def libdevice_exp(x):
    return f"libdevice.exp({x})"
 
```

Reply:
Very good point. Now that you mention it, in most cases it would be equivalent to just have a decorator that upcasts inputs and downcasts the output. I can try this out.

We might be able to inspect the decorator to automatically generate a test list as well, as @Skylion007 suggested.

Reply:
@blaine-rister I added tests here - https://github.com/pytorch/pytorch/pull/140057. we take all the opinfos for pointwise ops we have inductor lowerings of. let's follow up a bit more on other planning in workplace chat.

Reply:
I changed things to the decorator in 
6364feb. This seems to be working well, although a few of the test cases are failing. It seems like float16/bfloat16 dtypes might not be propagating through all those cases, meaning the codegen doesn't end up casting up. I'll follow up offline.